airflow/executors/base_executor.py

```diff
@@ -63,6 +63,14 @@ def __init__(self, parallelism: int = PARALLELISM):
             = OrderedDict()
         self.running: Set[TaskInstanceKey] = set()
         self.event_buffer: Dict[TaskInstanceKey, EventBufferValueType] = {}
+        self.job_id = None
+
+    def set_job_id(self, job_id):
+        """
+
+        @return:
+        """
+        self.job_id = job_id
```

Comment:
Why do we need a method for this (over just a "plain" `executor.job_id = X`)?

Reply:
So make this a setter function?

Reply:
Just make it a plain old property?

Reply:
@ashb already did :)